src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    public List<Machine> getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        if (type == Truck.class) {
+            return new TruckProducer().get();
+        } else {
+            return new ArrayList<>();
+        }
+
```

Comment:
I'd like to leave here several points:
- you are creating new producers each time the method is called.
- I'd recommend creating a map as a class field (using class as a key and producer as a value) and using this map instead of if-else
- let's use Collections.emptyList() in case we are going to return empty list